### PR TITLE
Change WorkspaceCleaner to look for bootstrap scripts as well as script log files

### PR DESCRIPTION
**Pull request by APErebus**
# Background

The `WorkspaceCleaner` is a background job that checks to make sure there aren't any orphaned workspaces hogging disk space. We discovered that for the kubernetes agent, this wasn't working because it was looking for the Script log (`Output.log`) to see if the workspace was orphaned (which we don't write for the k8s agent)

# Results

Change `IsUncompletedWorkspaceDirectory` to look for the bootstrap script _if not script log is found_, to determine if the workspace is "uncompleted".

We also change the `WorkspaceCleaner` to look for script log file first, then fallback on the bootstrap script if it doesn't exist

Shortcut story: [sc-77246]

# How to review this PR

Quality :heavy_check_mark:

# Pre-requisites

- [ ] I have read [How we use GitHub Issues](https://github.com/OctopusDeploy/Issues/blob/main/docs/CONTRIBUTING.internal.md) for help deciding when and where it's appropriate to make an issue.
- [x] I have considered informing or consulting the right people, according to the [ownership map](https://whimsical.com/ownership-map-NzbiD4HJyvhC9jNJNfS6TG).
- [x] I have considered appropriate testing for my change.

### CODE REVIEW

File: source/Octopus.Tentacle/Scripts/ScriptWorkspace.cs [hunk 1]
```diff
@@ -30,17 +30,16 @@ public class ScriptWorkspace : IScriptWorkspace
             fileSystem.EnsureDiskHasEnoughFreeSpace(workingDirectory);
         }
 
-        protected virtual string BootstrapScriptName => "Bootstrap.ps1";
+        const string BootstrapScriptFileName = "Bootstrap.ps1";
+        protected virtual string BootstrapScriptName => BootstrapScriptFileName;
 
         const string LogFileName = "Output.log";
-        public string LogFilePath => GetLogFilePath(WorkingDirectory);
+        public string LogFilePath => Path.Combine(WorkingDirectory, LogFileName);
```
Comment (LukeButters) [inline source/Octopus.Tentacle/Scripts/ScriptWorkspace.cs:37]: Why did this change?

Comment (APErebus) [reply source/Octopus.Tentacle/Scripts/ScriptWorkspace.cs:37]: The static `GetLogFilePath` wasn't needed outside of this class, so I removed the static method and changed this to use it's logic.

**Review by LukeButters — APPROVED** [review]

File: source/Octopus.Tentacle/Scripts/ScriptWorkspaceFactory.cs [hunk 2]
```diff
@@ -143,8 +143,20 @@ public List<IScriptWorkspace> GetUncompletedWorkspaces()
 
         static bool IsUncompletedWorkspaceDirectory(string workspaceDirectory)
         {
+            //first we look for an output log. If there is one, the workspace is uncompleted 
             var outputLogFilePath = ScriptWorkspace.GetLogFilePath(workspaceDirectory);
-            return File.Exists(outputLogFilePath);
+            if (File.Exists(outputLogFilePath))
```
Comment (flin-8) [inline source/Octopus.Tentacle/Scripts/ScriptWorkspaceFactory.cs:148]: Seems a bit unfortunate to have this check both here and in the workspace cleaner 🤔

Comment (APErebus) [reply source/Octopus.Tentacle/Scripts/ScriptWorkspaceFactory.cs:148]: yeah, I think the check in the workspace cleaner is a bit of a fallback option for edge cases

File: source/Octopus.Tentacle/Maintenance/WorkspaceCleaner.cs [hunk 3]
```diff
@@ -52,18 +52,18 @@ public async Task Clean(CancellationToken cancellationToken)
                     {
                         continue;
                     }
+                    
+                    //First look to see if there is a log file and if we can delete the workspace based on that
```
Comment (LukeButters) [inline source/Octopus.Tentacle/Maintenance/WorkspaceCleaner.cs:56]: Potentially the changes in https://github.com/OctopusDeploy/OctopusTentacle/pull/919 make this code clearer.

**Review by LukeButters — APPROVED** [review]

File: source/Octopus.Tentacle/Maintenance/WorkspaceCleaner.cs (hunk 3 above)
Comment (APErebus) [reply source/Octopus.Tentacle/Maintenance/WorkspaceCleaner.cs:56]: yeah, I think I'll pull your change in

Comment (APErebus) [reply source/Octopus.Tentacle/Maintenance/WorkspaceCleaner.cs]: cherry picked into this PR

**Pull request merged**